packages/tables/frontend/sirius-components-tables/src/table/column/useTableColumnSizing.ts

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2024 CEA LIST.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { UseTableColumnSizingValue } from './useTableColumnSizing.types';
+import { useEffect, useState } from 'react';
+import { MRT_ColumnSizingState } from 'material-react-table';
+import { GQLTable } from '../TableContent.types';
+import { useTableMutations } from '../../graphql/mutation/useTableMutation';
+
+export const useTableColumnSizing = (
+  editingContextId: string,
+  representationId: string,
+  table: GQLTable
+): UseTableColumnSizingValue => {
+  const [columnSizing, setColumnSizing] = useState<MRT_ColumnSizingState>({});
+  const { resizeColumn } = useTableMutations(editingContextId, representationId, table.id);
+
+  useEffect(() => {
+    for (const [columnName, columnSize] of Object.entries(columnSizing)) {
```

Comment:
if I have tens columns, this will fire tens HTTP requests when I resize a column? Why not have a mutation that can take all the data at once?

Reply:
As after each column resizing, I reset the local state to accept other values from other representations and it is physically possible to change only one value from the GUI. I can't have multiple entries in columnSizing.
That's why I hadn't taken into account passing a list